dynamodb/src/main/scala/zio/dynamodb/DynamoDBExecutorImpl.scala

```diff
@@ -317,34 +319,16 @@ private[dynamodb] final case class DynamoDBExecutorImpl private (dynamoDb: Dynam
     )
   }
 
-  private def doBatchGetItem(batchGetItem: BatchGetItem): ZIO[Clock, Throwable, BatchGetItem.Response] = {
-    def keysAndAttrsToTableGet(ka: KeysAndAttributes.ReadOnly): TableGet = {
-      val maybeProjectionExpressions = ka.projectionExpressionValue.map(
-        _.split(",").map(ProjectionExpression.$).toSet
-      ) // TODO(adam): Need a better way to accomplish this
-      val keySet =
-        ka.keysValue
-          .map(m => PrimaryKey(m.flatMap { case (k, v) => DynamoDBExecutorImpl.awsAttrValToAttrVal(v).map((k, _)) }))
-          .toSet
-
-      maybeProjectionExpressions.map(a => TableGet(keySet, a)).getOrElse(TableGet(keySet, Set.empty))
-    }
-    def tableItemsMapToResponse(tableItems: ScalaMap[String, List[ScalaMap[String, ZIOAwsAttributeValue.ReadOnly]]]) =
-      tableItems.foldLeft(MapOfSet.empty[TableName, Item]) {
-        case (acc, (tableName, list)) =>
-          acc ++ ((TableName(tableName), list.map(toDynamoItem)))
-      }
-
+  // TODO: Should be tail recursive
```

Comment:
I think having no tail recursion is OK here as the function returns an effect which makes it stack safe